src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2543,21 +2545,37 @@ private static bool WillResultBeAcceptable(bool isStale, IndexQueryBase<Blittabl
             return false;
         }
 
-        protected virtual unsafe long CalculateIndexEtag(bool isStale, DocumentsOperationContext documentsContext,
-            TransactionOperationContext indexContext)
+        protected virtual unsafe long CalculateIndexEtag(DocumentsOperationContext documentsContext,
+            TransactionOperationContext indexContext, QueryMetadata q, bool isStale)
         {
             var length = MinimumSizeForCalculateIndexEtagLength();
 
             var indexEtagBytes = stackalloc byte[length];
 
             CalculateIndexEtagInternal(indexEtagBytes, isStale, State, documentsContext, indexContext);
 
+            UseAllDocumentsEtag(documentsContext, q, length, indexEtagBytes);
```

Comment:
I'm not sure I follow: After we set all the values for the collections in the "indexEtagBytes", we then just override the first 16 bytes with some other values? I know that we are hashing it all at the end, but we do have higher chance for collision here, wouldn't we want to just have 16 more bytes for that usage?

Reply:
There is no need. We are overriding the lastDocumentEtag/lastTombstoneEtag from the 'first' collection that index is working on with lastDocumentEtag/lastTombstoneEtag for all docs. At the end this gives us needed uniqueness without the need of extending the buffer length.